### ISSUE TITLE
Mistake in convolution padding size of Denselayer

### ISSUE THREAD

**Issue by ZhangYuef** [xavysp/DexiNed#49]
Hi Xavier, thanks for your great job!

I found a little mistake in your model's PyTorch implement.

I think the second convolution padding size should be `2` in order to keep the input's last two dimension unchanged after the DenseLayer operation.  I think is should be
```
        self.add_module('relu1', nn.ReLU(inplace=True)),
        self.add_module('conv2', nn.Conv2d(out_features, out_features,
                                           kernel_size=3, stride=1, padding=2, bias=True)),
        self.add_module('norm2', nn.BatchNorm2d(out_features))
```

Btw, the script test for `model.py` has dimension error before this advise.

The code reference is [here](https://github.com/xavysp/DexiNed/blob/51f6b82f9b2a82fd2330283e79d0af1aeb25da88/DexiNed-Pytorch/model.py#L38), please correct me if I am wrong. Thanks : )

**Comment by xavysp** [comment]
> Hi Xavier, thanks for your great job!
> 
> I found a little mistake in your model's PyTorch implement.
> 
> I think the second convolution padding size should be `2` in order to keep the input's last two dimension unchanged after the DenseLayer operation. I think is should be
> 
> ```
>         self.add_module('relu1', nn.ReLU(inplace=True)),
>         self.add_module('conv2', nn.Conv2d(out_features, out_features,
>                                            kernel_size=3, stride=1, padding=2, bias=True)),
>         self.add_module('norm2', nn.BatchNorm2d(out_features))
> ```
> 
> Btw, the script test for `model.py` has dimension error before this advise.
> 
> The code reference is [here](https://github.com/xavysp/DexiNed/blob/51f6b82f9b2a82fd2330283e79d0af1aeb25da88/DexiNed-Pytorch/model.py#L38), please correct me if I am wrong. Thanks : )

Hi @ZhangYuef  you are right but I have added the padding in the 'conv1'
'''
self.add_module('conv1', nn.Conv2d(input_features, out_features,
                                           kernel_size=3, stride=1, padding=2, bias=True))
'''
I will update the DexiNed-Pytorch next week.
Cheers.

Xavier

**Comment by ZhangYuef** [comment]
> > Hi Xavier, thanks for your great job!
> > I found a little mistake in your model's PyTorch implement.
> > I think the second convolution padding size should be `2` in order to keep the input's last two dimension unchanged after the DenseLayer operation. I think is should be
> > ```
> >         self.add_module('relu1', nn.ReLU(inplace=True)),
> >         self.add_module('conv2', nn.Conv2d(out_features, out_features,
> >                                            kernel_size=3, stride=1, padding=2, bias=True)),
> >         self.add_module('norm2', nn.BatchNorm2d(out_features))
> > ```
> > 
> > 
> > Btw, the script test for `model.py` has dimension error before this advise.
> > The code reference is [here](https://github.com/xavysp/DexiNed/blob/51f6b82f9b2a82fd2330283e79d0af1aeb25da88/DexiNed-Pytorch/model.py#L38), please correct me if I am wrong. Thanks : )
> 
> Hi @ZhangYuef you are right but I have added the padding in the 'conv1'
> 
> ```
> self.add_module('conv1', nn.Conv2d(input_features, out_features,
>                                            kernel_size=3, stride=1, padding=2, bias=True))
> ```
> 
> I will update the DexiNed-Pytorch next week.
> Cheers.
> 
> Xavier

So it should has `padding=2` for both convolution layer here.

Thanks for your sharing @xavysp =)

**Comment by xavysp** [comment]
> > > Hi Xavier, thanks for your great job!
> > > I found a little mistake in your model's PyTorch implement.
> > > I think the second convolution padding size should be `2` in order to keep the input's last two dimension unchanged after the DenseLayer operation. I think is should be
> > > ```
> > >         self.add_module('relu1', nn.ReLU(inplace=True)),
> > >         self.add_module('conv2', nn.Conv2d(out_features, out_features,
> > >                                            kernel_size=3, stride=1, padding=2, bias=True)),
> > >         self.add_module('norm2', nn.BatchNorm2d(out_features))
> > > ```
> > > 
> > > 
> > > Btw, the script test for `model.py` has dimension error before this advise.
> > > The code reference is [here](https://github.com/xavysp/DexiNed/blob/51f6b82f9b2a82fd2330283e79d0af1aeb25da88/DexiNed-Pytorch/model.py#L38), please correct me if I am wrong. Thanks : )
> > 
> > 
> > Hi @ZhangYuef you are right but I have added the padding in the 'conv1'
> > ```
> > self.add_module('conv1', nn.Conv2d(input_features, out_features,
> >                                            kernel_size=3, stride=1, padding=2, bias=True))
> > ```
> > 
> > 
> > I will update the DexiNed-Pytorch next week.
> > Cheers.
> > Xavier
> 
> So it should has `padding=2` for both convolution layer here.
> 
> Thanks for your sharing @xavysp =)

No exactly, because the convolution for the skip connection was performed with a 1x1 kernel size , by the way, whata are you working on?

**Comment by ZhangYuef** [comment]
Hi @xavysp , I don't understand you totally. I am trying to run `python model.py` and find the dimension error.

**Comment by xavysp** [comment]
> Hi @xavysp , I don't understand you totally. I am trying to run `python model.py` and find the dimension error.

Did you update the last version of DexiNed-Pytorch, because I have solved there, its because of the padding in the upsampling layers

**Comment by ZhangYuef** [comment]
I got you. Solved in `commit a31ced957`

**Comment by ZhangYuef** [comment]
Thanks. I got you. Solved in `commit a31ced957`
